specs/executionAPISpec.md

```diff
@@ -0,0 +1,530 @@
+* [EFT-4] eth_filterTransaction **MUST** error with code -32000 when the `from` address does not have enough Ether to pay for the transaction.
+* [EFT-5] eth_filterTransaction **MUST** allow both `data` and `input` to be used for contract creation,
+* [EFT-6] eth_filterTransaction **MUST** error with code -32000 when the `data` and `input` both specified and not equal
+* [EFT-7] eth_filterTransaction **MUST** error with code -32000 when `data` or `input` or `to` is not specified.
+* [EFT-8] eth_filterTransaction **MUST** error with code -3200 when `data` or `input` provided caused an EVM error when the `gas` is not specified, otherwise it is ignored.
+## <p id="createAccessList">eth_createAccessList</p>
+* [ECAL-1] eth_createAccessList **MUST** return an [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) access list biased off the given `transaction` and estimated gas cost when using the access list in the `transaction`.
+* [ECAL-2] eth_createAccessList **MUST** use "latest" when the `defaultBlockParameter` is not specified.
+* [ECAL-3] eth_createAccessList **MUST** use the estimated gas when the `gas` is not specified.
+* [ECAL-4] eth_createAccessList **MUST** error with code -32000 when `gas` is too low.
+* [ECAL-5] eth_createAccessList **MUST** error with code -32000 when the `from` address does not have enough Ether to pay for the transaction.
+## <p id="estimateGas">eth_estimateGas</p>
+* [EEG-1] eth_estimateGas **MUST** return the estimated amount of gas the given `transaction` will take to execute.
+* [EEG-2] eth_estimateGas **MUST** error with code -32000 when the `from` address has insufficient Ether to execute the given `transaction`.
+* [EEG-3] eth_estimateGas **MUST NOT** check if the `from` account has sufficient funds when estimating contract deployment.
+* [EEG-4] eth_estimateGas **MUST** use 0x0000000000000000000000000000000000000000 for `from` when it is not specified.
+* [EEG-5] eth_estimateGas **MUST** error with code -32000 when estimating a contract creation that causes an error within the EVM.
+## <p id="signTransaction">eth_signTransaction</p>
+* [ESNT-1] eth_signTransaction **MUST** return both the signed encoded and signed unencoded format of the `transaction` given.
+* [ESNT-2] eth_signTransaction **MUST NOT** use given transaction `type` for transaction type, **MUST** determine transaction type from given transaction parameters.
+* [ESNT-3] eth_signTransaction **MUST** return encoded legacy transactions in the RLP format.
+* [ESNT-4] eth_signTransaction **MUST** return encoded [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) transactions as the transaction `type` concatenated with the transaction type's defined encoding method.
+* [ESNT-5] eth_signTransaction **MUST** use 0x0000000000000000000000000000000000000000 as default `from` address when `from` is null or not specified.
+* [ESNT-6] eth_signTransaction **MUST** error with code -32000 when the `from` address is locked.
```

Comment:
Side note: Why does *everything* return error code -32000?  It feels like we should be returning useful error codes.  Is this just an issue of how current clients all use the same error code, so now we are stuck with uselessness?

Reply:
Right! I was wondering the same thing. Most errors do have unique error messages though.